stl/inc/xstring

```diff
@@ -2528,13 +2612,30 @@ public:
         _Elem* _Ptr             = _My_data._Bx._Buf;
         auto&& _Alproxy         = _GET_PROXY_ALLOCATOR(_Alty, _Getal());
         _Container_proxy_ptr<_Alty> _Proxy(_Alproxy, _My_data); // throws
-        if (_New_capacity < _New_size) {
-            _New_capacity           = _Calculate_growth(_New_size, _BUF_SIZE - 1, max_size());
+
+#ifdef __cpp_lib_is_constant_evaluated
+        if (_STD is_constant_evaluated()) {
+            _New_capacity           = _BUF_SIZE; // SSO disabled in constexpr context
             const pointer _Fancyptr = _Getal().allocate(_New_capacity + 1); // throws
             _Ptr                    = _Unfancy(_Fancyptr);
             _Construct_in_place(_My_data._Bx._Ptr, _Fancyptr);
+        } else
+#endif // __cpp_lib_is_constant_evaluated
+        {
+            if (_New_capacity < _New_size) {
+                _New_capacity           = _Calculate_growth(_New_size, _BUF_SIZE - 1, max_size());
+                const pointer _Fancyptr = _Getal().allocate(_New_capacity + 1); // throws
+                _Ptr                    = _Unfancy(_Fancyptr);
+                _Construct_in_place(_My_data._Bx._Ptr, _Fancyptr);
+            }
         }
 
+#ifdef __cpp_lib_is_constant_evaluated
+        if (_STD is_constant_evaluated()) { // Copying into uninitialized memory is technically UB, so fill it once
+            _Traits::assign(_Ptr, _New_capacity + 1, _Elem());
+        }
+#endif // __cpp_lib_is_constant_evaluated
+

```

Comment:
This here was one possible bug, where I failed to account for reallocation during constant evaluation.

Still unsure whether that could lead to those failures. CI will tell...